app/viewModels/DepartureStatusViewModel.scala

```diff
@@ -162,9 +138,11 @@ object DepartureStatus {
       )
   }
 
-  private def departureXmlNegativeAcknowledgement: PartialFunction[Departure, DepartureStatus] = {
-    case departure if departure.status == DepartureSubmittedNegativeAcknowledgement =>
-      DepartureStatus(
+  private def departureXmlNegativeAcknowledgement: PartialFunction[Departure, DepartureStatusViewModel] = {
+    case departure
+        if departure.currentStatus == XMLSubmissionNegativeAcknowledgement &&
```

Comment:
Not to get too OO, but would it make sense to move this bit of logic into the departure view model as a method (the double "if")? And the same for the other ack case as well.

Reply:
Sort of hide that detail

Reply:
Approved anyway, but something to consider.